File: frontend/src/components/types/products/product.ts
export interface ProductsProps {
  heading?: string;
  products?: Product[];
}
export interface Product {
  name: string;
  price: number;
  pictureUrl: string;
  description: string;
  id: string;
}

export interface GetAllProducts {
  result: Product[];
  itemLength: number;
}

File: backend/Models/Product.cs
namespace backend.Models;

public class Product : BaseModel
{
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public double Price { get; set; }
    public string PictureUrl { get; set; } = null!;
    public string PublicId { get; set; } = null!;
    public Category Category { get; set; } = null!;
    public Guid CategoryId { get; set; }

}
